src/quo2/components/tags/context_tag/view.cljs

```diff
@@ -1,105 +1,206 @@
-      (when channel-name
-        [:<>
-         [icons/icon
-          :i/chevron-right
-          {:color (style/context-tag-icon-color blur?)
-           :size  16}]
-         [text/text text-props (str "# " channel-name)]])]]))
-
-(defn user-avatar-tag
-  [params username photo]
-  [context-tag params photo username])
-
-(defn audio-tag
-  [duration params]
-  [base-tag (merge {:style style/audio-tag-container} params)
-   [rn/view {:style style/audio-tag-icon-container}
-    [icons/icon
-     :i/play
-     {:color style/audio-tag-icon-color
-      :size  12}]]
+  (str (subs pk 0 5) "..." (subs pk (- (count pk) 3))))
+
+(defn- address-tag
+  [{:keys [theme size address]}]
+  [rn/view {:style (style/address size)}
    [text/text
-    {:weight :medium
-     :size   :paragraph-2
-     :style  {:margin-left 4
-              :color       (style/audio-tag-text-color (:override-theme params))}}
-    duration]])
-
-(defn community-tag
-  [avatar community-name {:keys [override-theme] :as params}]
-  [context-tag
-   (merge {:style                style/community-tag
-           :text-style           (style/community-tag-text override-theme)
-           :text-container-style style/community-tag-text-container
-           :ellipsize-text?      true}
-          params)
-   avatar
-   community-name])
+    {:style  (style/text theme)
+     :weight :monospace ;; TODO: fix this style
+     :size   (if (= size 24) :paragraph-2 :paragraph-1)}
+    (trim-public-key address)]])
+
+(defn- icon-tag
+  [{:keys [theme size icon blur? context]}]
+  [rn/view {:style (style/icon size)}
+   [icons/icon icon
+    {:color (style/context-tag-icon-color theme blur?)
+     :size  (if (= size 24) 12 20)}]
+   [rn/view {:style (style/icon-spacing size)}
+    [text/text
+     {:style  (style/text theme)
+      :weight :medium
+      :size   (if (= size 24) :paragraph-2 :paragraph-1)}
+     context]]])
+
+(defn- view-internal
+  [{:keys [theme type size state blur? customization-color profile-picture full-name users
+           group-name token-logo amount token-name network-logo network-name networks
+           account-name emoji collectible collectible-name collectible-number duration]
+    :or   {customization-color :blue
+           type                :default
+           state               :default}
+    :as   props}]
+  [rn/view {:style (style/container theme type size state blur? customization-color)}
+   (case type
+     :default
+     [tag-skeleton {:theme theme :size size :text full-name}
+      [user-avatar/user-avatar
+       {:full-name         full-name
+        :profile-picture   profile-picture
+        :size              (if (= size 24) :xxs 28)
+        :status-indicator? false}]]
+
+     :multiuser
+     [preview-list/preview-list {:type :user :size 20}
+      users]
+
+     :multinetwork ;; TODO: use multinetwork preview-list type when available
+     [preview-list/preview-list {:type :user :size 20}
+      (map #(hash-map :profile-picture %) networks)]
```

Comment:
the multinetwork component is not available, so we should use it here when it's ready